### PR TITLE
Add button to launch the Crash Reporter from the pause menu

### LINKED ISSUE #2577: Add a button to display the CrashReporter in the main menu
### Description

When Terasology encounters a fatal exception, a lightweight Crash Reporter application is launched, allowing users to report the error using various methods - upload the crash log to Pastebin or Google Drive, file a GitHub issue and so on.

Crash Reporter screenshot

However, non-fatal errors and bugs cannot be easily reported using the Crash Reporter interface - the user will have to manually locate the game log then report it using whatever method they can think of. A button that'll open the Crash Reporter under normal working conditions would help report these non-critical issues.

### Definition of 'Done'

- A button is added to the game's pause menu that opens the Crash Reporter interface, showing the game's current log.
- _[Optional but nice to have]_ The Crash Reporter itself is reworked to have a "non-critical" mode that changes several strings and interface elements to remove any mentions of a crash having occurred (e.g. in the screenshot above, "A fatal error occurred" is changed to "Current game log" and the red warning icon is changed to something more neutral). This mode should be used when the Crash Reporter is launched manually.

### Required skills

The issue involves working with the game's codebase and requires intermediate level Java skills.

### Where to start?

- The crash reporter can be launched using `CrashReporter.report(Throwable throwable, Path logPath);`.
  - The exception being thrown should be a dummy, since no critical exception actually occurs.
  - The path to the game log can be retrieved via `LoggingContext.getLoggingPath()`.
- Adding the button to the pause menu requires modifying its asset file ([pauseMenu.ui](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/resources/assets/ui/ingame/pauseMenu.ui)), as well as the code-behind class ([`PauseMenu`](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/rendering/nui/layers/ingame/PauseMenu.java)). For a quick overview on working with interface screens, check out the [NUI Quick Start](https://github.com/Terasology/TutorialNui/wiki/Quick-Start) tutorial!
- The optional item will require working on the Crash Reporter itself, which is located in an auxiliary repository: https://github.com/MovingBlocks/CrashReporter.

### PR DESCRIPTION
**Pull request by segfault802**
### Contains

Fixes #2577 

screen shot 2016-10-30 at 8 38 15 pm
### How to test

Load up a game
Press **Esc** to open the pause menu
There is a new button which brings up the crash reporter with the current game log
### Outstanding before merging
- [x] Decide if the button should go anywhere else
- [ ] Decide what to do about Crash Reporter dependency issue (see comment on #2577)

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/build.gradle [hunk 1]
```diff
@@ -132,6 +132,9 @@ dependencies {
 
     // In addition to all the above the dev source set also needs to depend on what gets compiled in main
     devCompile sourceSets.main.output
+
+    // Dependency on CrashReporter
+    compile group: 'org.terasology.crashreporter', name: 'cr-terasology', version: '4.0.0'
```
Comment (rzats) [inline engine/build.gradle]: Can you try using the fancier setup method from https://github.com/MovingBlocks/Terasology/blob/develop/facades/PC/build.gradle#L43-L58? Might fix the issue you've mentioned in the... issue 😄

**Comment by Cervator** [comment]
As for the button placement it looks great next to the Dev Tools. I would like to also see it in the main menu itself (bottom right corner as well) if not too tricky :-)

**Comment by segfault802** [comment]
Added the button to the main menu also, same location.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by skaldarnar** [comment]
@rzats @Cervator can this be merged with manual adaption of the CR dependency? Buttons are all in place as requested.

**Comment by rzats** [comment]
@skaldarnar: if you know how to fix the dependency issue, go ahead 🙂

File: engine/build.gradle (hunk 1 above)
Comment (skaldarnar) [inline engine/build.gradle]: @Cervator any news on this?

**Comment by flo** [comment]
Is anyone working on solving the dependency issue? Else I suggest we close this PR for now as sad as it is.

**Comment by iojw** [comment]
@flo I've likely got a fix for the dependency issue, just need time to test it out.

**Comment by flo** [comment]
ok cool

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by iojw** [comment]
Is anyone able to reproduce the crashing on start problem? I've tested this locally with the engine build.gradle updated to use the CR source conditionally. I've also confirmed that both `com.google.guava:guava-jdk5:13.0` and `com.google.guava:guava:19.0` are getting pulled in by gradle when launching the game. The game appears to load fine and CrashReporter launches without any issues from both the main menu screen and pause menu.

Since `guava-jdk5` does not seem to be maintained anymore, we can simply exclude `guava-jdk5` from the cr-core build.gradle compile dependencies and update it to use vanilla guava instead, like so:

```    
compile ('com.google.apis:google-api-services-drive:v2-rev193-1.20.0'){ exclude module: 'guava-jdk5' }
compile group: 'com.google.guava', name: 'guava', version: '19.0'
```

**Comment by iojw** [comment]
Yup, I've confirmed that the above change fixed the issue. 

Received the following error running the game after cleaning my workspace once, which indicates that Guava is present multiple times on the classpath:
```
java.lang.NoSuchMethodError: com.google.common.collect.Queues.newLinkedBlockingDeque()Ljava/util/concurrent/LinkedBlockingDeque;
	at org.terasology.engine.subsystem.lwjgl.LwjglGraphics.<init>(LwjglGraphics.java:101)
	at org.terasology.engine.Terasology.populateSubsystems(Terasology.java:371)
	at org.terasology.engine.Terasology.main(Terasology.java:129)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

The above error no longer occurs after the cr-core build.gradle is updated. I've submitted a PR at MovingBlocks/CrashReporter#34 to fix this issue. After it has been merged, the engine build.gradle can also be updated to use the CR source conditionally as mentioned above and then this PR will be good to go 😄.

**Comment by msteiger** [comment]
Thanks, @iojw - I've merged your PR. See my comment there.

**Comment by Cervator** [comment]
Released the CR fix then bumped the version dependency on top of this PR, as well as adding the conditional source vs binary magic. Merged! :-)

Tested out both as source and binary. There are some quirks and follow-ups we should handle, but those are not problems with this PR.

Thanks @segfault802! Thanks @iojw! Thanks reviewers! :-)

**Pull request merged**